### ISSUE TITLE
go vet fails

### ISSUE THREAD

**Issue by gerhard** [streadway/amqp#231]
go vet fails:

```sh
$ go vet
connection.go:188: assignment copies lock value to c: crypto/tls.Config contains sync.Once contains sync.Mutex
exit status 1
$ go version
go version go1.7.5 darwin/amd64
```

**Comment by streadway** [comment]
This still needs some discussion. In the meantime, run `pre-commit` which has this exception included.

**Comment by streadway** [comment]
To fix this, https://github.com/streadway/amqp/blob/master/connection.go#L189-L194 would need to mutate the provided `tls.Config`.

The `tls.Config` mutex is only used for server configurations, so the `go vet` error wouldn't affect this library.

I'm ok mutating instead of copying the passed `amqp.Config.TLSClientConfig` in `DialConfig` in order to satisfy this vet message when the hostname isn't provided in `TLSClientConfig`.

**Comment by streadway** [comment]
Go 1.8 will introduce https://beta.golang.org/pkg/crypto/tls/#Config.Clone which accomplishes the same thing as the current code without the vet error.

**Comment by gerhard** [comment]
Happy to wait for Go 1.8 (due next month, 2017.02) and fix this when we switch to 1.8. Does this mean that we'll drop support for Go < 1.8 in master?

**Comment by gerhard** [comment]
Interface type assertion on Config.Clone to support all Go versions.

**Comment by gerhard** [comment]
Closed in #238
